app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java

```diff
@@ -45,18 +47,27 @@
   private final Billing billing;
   private final SharedPreferences sharedPreferences;
   private final PackageManager packageManager;
+  private final BackupInteractContract backupInteract;
 
   public InAppPurchaseInteractor(AsfInAppPurchaseInteractor asfInAppPurchaseInteractor,
       BdsInAppPurchaseInteractor bdsInAppPurchaseInteractor,
       ExternalBillingSerializer billingSerializer, AppcoinsRewards appcoinsRewards, Billing billing,
-      SharedPreferences sharedPreferences, PackageManager packageManager) {
+      SharedPreferences sharedPreferences, PackageManager packageManager,
+      BackupInteractContract backupInteract) {
     this.asfInAppPurchaseInteractor = asfInAppPurchaseInteractor;
     this.bdsInAppPurchaseInteractor = bdsInAppPurchaseInteractor;
     this.billingSerializer = billingSerializer;
     this.appcoinsRewards = appcoinsRewards;
     this.billing = billing;
     this.sharedPreferences = sharedPreferences;
     this.packageManager = packageManager;
+    this.backupInteract = backupInteract;
+  }
+
+  public Single<NotificationNeeded> incrementAndValidateNotificationNeeded() {
+    return getWalletAddress().flatMap(s -> backupInteract.updateWalletPurchasesCount(s)
```

Comment:
getWalletAddress can be private as it's not being used outide of this class (which is strange). So I think we should call asfInAppPurchaseInteractor.getWalletAddress(); directly here